src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -605,6 +610,8 @@ private static void AnalyzeMessageCore(string message, IArgumentOperation argume
                     fixedMessage = endsWithPeriod ? RemoveTrailingPeriod(fixedMessage) : fixedMessage + ".";
                 }
 
+                fixedMessage = hasLeadingOrTrailingWhitespaces ? RemoveLeadingAndTrailingWhitespaces(fixedMessage) : fixedMessage;
```

Comment:
I think this might have to move ahead to just after we fix line returns but before the above if statement.